callstack/linaria
# Use workspace root as a base for cache folder

**Summary**


Currently, `process.cwd()` is used as a base directory for cache file resolution. The problem with this is that we run into issues when there are components in yarn workspaces because they are outside the cwd. This leads to cache files being generated outside `.linaria-cache`.

This PR detects the yarn workspace root via [https://www.npmjs.com/package/find-yarn-workspace-root](find-yarn-workspace-root) and uses it if it exists. The PR is open for suggestions :).

**Test plan**


1. Create two yarn workspaces, one with the app and one that exports a component.
1. Add a css`foo` expression to the component.
1. Import the component workspace in the app workspace.
1. There should be a `component.linaria.css` file generated outside the `.linaria-cache` folder.
